Content.Shared/Body/Part/BodyPartComponent.cs

```diff
@@ -1,42 +1,82 @@
 ﻿using Content.Shared.Body.Components;
 using Content.Shared.Body.Organ;
 using Content.Shared.Body.Systems;
+using Robust.Shared.Containers;
 using Robust.Shared.GameStates;
+using Robust.Shared.Serialization;
 
 namespace Content.Shared.Body.Part;
 
-[RegisterComponent, NetworkedComponent]
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState(true)]
 [Access(typeof(SharedBodySystem))]
 public sealed partial class BodyPartComponent : Component
 {
-    [DataField("body")]
+    [DataField("body"), AutoNetworkedField]
     public EntityUid? Body;
 
-    // This inter-entity relationship makes be deeply uncomfortable because its probably going to re-encounter all of the
-    // networking issues that containers and joints have.
-    // TODO just use containers. Please.
-    // Do not use set or get data from this in client-side code.
-    public BodyPartSlot? ParentSlot;
+    [AutoNetworkedField, ViewVariables] public EntityUid? Parent;
 
-    // Do not use set or get data from this in client-side code.
-    [DataField("children")]
-    public Dictionary<string, BodyPartSlot> Children = new();
-
-    // See all the above ccomments.
-    [DataField("organs")]
-    public Dictionary<string, OrganSlot> Organs = new();
+    [AutoNetworkedField, ViewVariables] public string? AttachedToSlot = null;
 
-    [DataField("partType")]
+    [DataField("partType"), AutoNetworkedField]
     public BodyPartType PartType = BodyPartType.Other;
 
     // TODO BODY Replace with a simulation of organs
     /// <summary>
     ///     Whether or not the owning <see cref="Body"/> will die if all
     ///     <see cref="BodyComponent"/>s of this type are removed from it.
     /// </summary>
-    [DataField("vital")]
+    [DataField("vital"), AutoNetworkedField]
     public bool IsVital;
 
-    [DataField("symmetry")]
+    [DataField("symmetry"), AutoNetworkedField]
     public BodyPartSymmetry Symmetry = BodyPartSymmetry.None;
+
+    [AutoNetworkedField(CloneData = true), ViewVariables]
+    public Dictionary<string, BodyPartSlot> Children = new();
+
+
+    [AutoNetworkedField(CloneData = true), ViewVariables]
+    public Dictionary<string, OrganSlot> Organs = new();
+
+    [ViewVariables]
+    public List<ContainerSlot> BodyPartSlots
+    {
+        get
+        {
+            List<ContainerSlot> temp = new();
+            foreach (var (_,slotData) in Children)
+            {
+                temp.Add(slotData.Container);
+            }
+            return temp;
+        }
+    }
+
+    [ViewVariables]
+    public List<ContainerSlot> OrganSlots
+    {
+        get
+        {
+            List<ContainerSlot> temp = new();
+            foreach (var (_,slotData) in Organs)
+            {
+                temp.Add(slotData.Container);
+            }
+            return temp;
+        }
+    }
```

Comment:
i dont like that these are getters, imo should be a method so any systems calling into it understand that this will incur a new allocation for getting the list and what not

Reply:
not really sure why these need to exist anyhow when you can just iterate over organs/children.values and get the container there

Reply:
this is solely for debug (VV)

Reply:
can the property names get changed to something like `OrganSlotsVV` then? i think they can be made private too and still work

Reply:
prob add a xmldoc too

Reply:
Sure good idea